### ISSUE TITLE
Dependency versions are overly restrictive

### ISSUE THREAD

**Issue by shepmaster** [PistonDevelopers/image-tiff#21]
Using a `*` forces downstream users to be unable to use newer versions of crates. It is usually suggested to avoid a version operator at all. Currently, this crate has:

https://github.com/PistonDevelopers/image-tiff/blob/4e3b36598d97fe0096f4696495f3f2af8f4f0148/Cargo.toml#L18-L21

Instead, I would expect this to be written as:

```toml
byteorder = "1.2"
lzw = "0.10"
num-derive = "0.2"
num-traits = "0.2"
```

This would allow for byteorder 1.2.x and 1.3.x, a semver-compatible version. It would **not** allow num-derive 0.3 (which is not semver-compatible with 0.2).

See [the Cargo doc for details on the operators](https://doc.rust-lang.org/stable/cargo/reference/specifying-dependencies.html).

The playground uses the newest version of crates. The image crate is popular enough to be included on the playground and it depends on tiff. This version restriction makes it so that tiff cannot be installed at the same time as byteorder 1.3. Until this issue is resolved, I will be preventing image (and thus tiff) from being available in the playground.

Thanks for listening!

**Comment by shepmaster** [comment]
> Instead, I would expect this to be written as:
> 
> ```toml
> byteorder = "1.2"
> ```

Personally, I'd try to use `byteorder = "1"` and set up a CI instance using nightly's `-Zminimal-versions` flag to ensure that your code is compatible with the widest possible version range, in order to be used by as many projects as possible.

You can see an example of a wide version range on rand [in my project](https://github.com/shepmaster/twox-hash/blob/master/Cargo.toml#L17) and an example of [testing the early versions in iovec](https://github.com/carllerche/iovec/blob/master/.travis.yml#L14-L17).

**Comment by shepmaster** [comment]
> Cargo treats "x.y.z" as "^x.y.z".

Yes, which is why I suggest leaving off the operator.

> should use full "x.y.z".

There's no benefit, unless you **need** to specify the patch version. From [the linked documentation](https://doc.rust-lang.org/stable/cargo/reference/specifying-dependencies.html#caret-requirements):

> ```
> ^1.2.3 := >=1.2.3 <2.0.0
> ^1.2 := >=1.2.0 <2.0.0
> ^1 := >=1.0.0 <2.0.0
> ```

**Comment by shepmaster** [comment]
@bvssvni please feel free to ping me when a new version of tiff is released and I can add it back to the playground!

**Comment by CryZe** [comment]
I just ran into this as well. So I'm waiting on this to be published too I guess.

**Comment by CryZe** [comment]
Apparently 0.2.1 is indeed published, but cargo refuses to select 0.2.1 if you also have byteorder 1.3.1:
https://i.imgur.com/0pvwHS2.png
